lib/ansible/modules/crypto/openssh_keypair.py

```diff
@@ -202,6 +205,17 @@ def _check_state():
 
         if _check_state():
             proc = module.run_command([module.get_bin_path('ssh-keygen', True), '-lf', self.path])
+            if not proc[0] == 0:
+                if os.path.isdir(self.path):
+                    module.fail_json(msg='%s is a directory. Please specify a path to a file.' % (self.path))
+
+                if os.access(self.path, os.W_OK) and not self.force:
```

Comment:
I would simply replace this `if` block by `return False`. If the key is broken, it should simply be regenerated.